rpcs3/Emu/RSX/Program/FragmentProgramDecompiler.cpp

```diff
@@ -801,6 +805,22 @@ std::string FragmentProgramDecompiler::BuildCode()
 	}
 
 	std::stringstream OS;
+
+	if (!m_is_valid_ucode)
+	{
+		// If the code is broken, do not compile. Simply NOP main and write empty outputs
+		insertHeader(OS);
+		OS << "\n";
+		OS << "void main()\n";
+		OS << "{\n";
+		OS << "#if 0\n";
+		OS << main << "\n";
+		OS << "#endif\n";
```

Comment:
stupid question: why do we need this part if it's ifdefed out anyway?

Reply:
Analysis. It will show the bad code in logged shaders andvrdc (ogl)

Reply:
Ah ok. Thx